record/record_test.go

```diff
@@ -312,3 +313,137 @@ func TestStaleWriter(t *testing.T) {
 		t.Fatalf("stale write #1: unexpected error: %v", err)
 	}
 }
+
+func TestBasicReadRecovery(t *testing.T) {
+	buf := new(bytes.Buffer)
+	w := NewWriter(buf)
+
+	for i := 0; i < 3; i++ {
+		if wRec, err := w.Next(); err != nil {
+			t.Fatal(err)
+		} else {
+			if _, err = wRec.Write([]byte(strings.Repeat(string(rune(97+i)), blockSize-headerSize))); err != nil {
```

Comment:
Effectively did this. Used strings.Repeat but then casted to a []byte, instead of casting the input string and then using bytes.Repeat.